simtools/model/mirrors.py

```diff
@@ -136,10 +173,18 @@ def get_single_mirror_parameters(self, number):
         """
 
         def get_mirror_table_mask():
-            if isinstance(self.mirror_table["mirror_id"][0], np.str_):
-                mask = self.mirror_table["mirror_id"] == "id=" + str(number)
-            if isinstance(self.mirror_table["mirror_id"][0], np.int32):
-                mask = self.mirror_table["mirror_id"] == number
+            if isinstance(self.mirror_table["mirror_panel_id"][0], np.str_):
+                mask = (
+                    np.array(
+                        [
+                            "".join(filter(str.isdigit, string))
+                            for string in self.mirror_table["mirror_panel_id"]
+                        ]
+                    )
+                    == f"{number}"
+                )
+            if isinstance(self.mirror_table["mirror_panel_id"][0], np.int32):
```

Comment:
Actually, I realise now (after writing the comment above regarding the string), that we shouldn't do this at all. The id should always be a number. That is, when you read a table, you should cast the id to be a number, even if it is written as a string in the file.